org.lflang/src/org/lflang/generator/c/CCmakeGenerator.java

```diff
@@ -0,0 +1,206 @@
+        
+        cMakeCode.append("set(LF_MAIN_TARGET "+executableName+")\n");
+        cMakeCode.append("# Declare a new executable target and list all its sources\n");
+        cMakeCode.append("add_executable( ${LF_MAIN_TARGET} "+String.join("\n", sources)+" ${LF_PLATFORM_FILE} "+
+                           String.join("\n", additionalSources)+")\n");
+        cMakeCode.append("\n");
+
+        if (targetConfig.threads != 0 || targetConfig.tracing != null) {
+            // If threaded computation is requested, add a the threads option.
+            cMakeCode.append("# Find threads and link to it\n");
+            cMakeCode.append("find_package(Threads REQUIRED)\n");
+            cMakeCode.append("target_link_libraries( ${LF_MAIN_TARGET} Threads::Threads)\n");
+            cMakeCode.append("\n");
+            
+            // If the LF program itself is threaded or if tracing is enabled, we need to define
+            // NUMBER_OF_WORKERS so that platform-specific C files will contain the appropriate functions
+            cMakeCode.append("# Set the number of workers to enable threading\n");
+            cMakeCode.append("target_compile_definitions( ${LF_MAIN_TARGET} PUBLIC NUMBER_OF_WORKERS="+targetConfig.threads+")\n");
+            cMakeCode.append("\n");
+        }
+        
+        boolean CPPRequested = false;
+        if (targetConfig.compiler != null) {
+            if (targetConfig.compiler.equals("g++")) {
+                CPPRequested = true;
+                // Interpret this as the user wanting their .c programs to be treated as
+                // C++ files. 
+                // First enable the CXX language
+                cMakeCode.append("enable_language(CXX)\n");
+                cMakeCode.append("set(CMAKE_CXX_FLAGS \"${CMAKE_CXX_FLAGS} -Wno-write-strings\")\n");
+                // We can't just simply use g++ to compile C code. We use a 
+                // specific CMake flag to set the language of all .c files to C++.
+                for (String source: sources) {
+                    cMakeCode.append("set_source_files_properties( "+source+" PROPERTIES LANGUAGE CXX)\n");
+                }
+                // Also convert any additional sources
+                for (String source: additionalSources) {
+                    cMakeCode.append("set_source_files_properties( "+source+" PROPERTIES LANGUAGE CXX)\n");
+                }
+                cMakeCode.append("set_source_files_properties(${LF_PLATFORM_FILE} PROPERTIES LANGUAGE CXX)\n");
+            } else {
+                cMakeCode.append("set(CMAKE_C_COMPILER "+targetConfig.compiler+")\n");
+            }
+            
+            // cMakeCode.append("set(CMAKE_CXX_COMPILER "+targetConfig.compiler+")\n");
+        }
+        
+        // Set the compiler flags
+        // We can detect a few common libraries and use the proper target_link_libraries to find them            
+        for (String compilerFlag : targetConfig.compilerFlags) {
```

Comment:
This looks a bit dangerous to me. I think it would be better to forbid the usage of the compiler flags target property if cmake is enabled and use `cmake-include` instead. Note that the default case in the code below is not portable and would only work with gcc/g++ or clang.

Reply:
Ah, good point. I would be hesitant to remove this completely though (i.e., in the validator) because of backwards compatibility. If CMake was enabled by default, but wasn't installed for example, we might want to revert back to using the native gcc compiler. Using `cmake-include` will not help in that case. I will add a warning here for now.